core/src/main/java/org/bitcoinj/wallet/CoinSelection.java

```diff
@@ -33,8 +34,18 @@ public class CoinSelection {
     public final Coin valueGathered;
     public final Collection<TransactionOutput> gathered;
 
+    public CoinSelection(Collection<TransactionOutput> gathered) {
+        this(sumOutputValues(gathered), gathered);
+    }
+
     public CoinSelection(Coin valueGathered, Collection<TransactionOutput> gathered) {
         this.valueGathered = valueGathered;
         this.gathered = gathered;
     }
```

Comment:
I've wondered in the past if this constructor can be eliminated. In some use cases you already have the sum collected, so recalculating for a long `List` of `TransactionOutput` feels wasteful, but maybe simplicity is more important than optimization.

Reply:
Yes, that's what I'm thinking in https://github.com/bitcoinj/bitcoinj/pull/2555#issuecomment-1188440661 too.

Reply:
I just noticed this comment https://github.com/bitcoinj/bitcoinj/pull/2555#issuecomment-1188440661. So maybe we should deprecate the two-arg constructor.

Reply:
If you look at PR #2412 it avoids the garbage collection hit and creates a single new `Coin` for the total:

```
    public static Coin sum(Collection<TransactionOutput> outputs) {
        return Coin.ofSat(sumSats(outputs));
    }

    public static long sumSats(Collection<TransactionOutput> outputs) {
        return outputs.stream()
                .mapToLong(output -> output.value)
                .reduce(0, LongMath::checkedAdd);
    }
```